app/controllers/ems_cloud_controller.rb

```diff
@@ -35,6 +35,10 @@ def show_link(ems, options = {})
     ems_path(ems.id, options)
   end
 
+  def generate_options
+    {:association => 'storage_managers'}
+  end
```

Comment:
I understand what you are doing here, but don't get why you named the method this way. In the context of `EmsCloudController` the name `generate_options` hardly relates to what this method is really returning.

Reply:
The name `generate_options` actually came as a suggestion from the reviewers, I believe in this PR - https://github.com/ManageIQ/manageiq-ui-classic/pull/1553

Reply:
Does the fact that it came from the reviewers make it a better name now that you at the context where it is used?

Reply:
Well, the method is about `options` that we set on a case-by-case basis.
Perhaps `set_options` would have been more appropriate, but rubocop would complain about the `set` part.
I really do not mind `generate_options`, it *does* convey the intent of the method.
Do you have a better suggestion?

Reply:
Actually I think that we just remove it.

Once we merge https://github.com/ManageIQ/manageiq-ui-classic/pull/2383 we will not need to pass anything through the session and there should be no need for special hack such as this one.

I need to collect all the issues that we have with GTLs in one place so that we can make sure we fix it all.

Reply:
I would like to mention here that  `generate_options` is not just for setting `options` in the session, but it is also used for setting `options` that `report_data` will consume.

After #2383 merge, even if we remove `generate_options`, the logic to set the `options` still needs to persist for `report_data`. 
I assume `@report_data_additional_options` will continue with that trend of setting `options` for `report_data`

Reply:
The other way round. Code will set all the options and that will be passwed through the @report_data_additional_options to the GTL grid component.

The GTL component will fire requests to the `/report_data` endpoint and from the request (and nothing else) `/report_data` will reconstruct the @report_data_additional_options so that `/report_data` can send the correct data.

This means that ANY magic will be happening BEFORE rendering the partial with the grid. After that NO magic at all. After that we will be (mostly) stateless.

This means that we will be able to cleanup the magic that happens before the rendering w/o breaking what `/report_data` returns.

Reply:
So no, the `generate_options logic` must not persist for `report_data` ;-)

Reply:
Alright. So when you say "Code will set all the options...", which code specifically would do it?

Reply:
> which code specifically would do it?

Code that is executed prior to accessing `/report_data` before the partial with the form is rendered.

`/report_data` will then receive all the options in the HTTP POST request.

I hope we will remove any exceptions that are now in `process_params_options` because all the options will be set in the HTTP request that calls `nested_list` or similar. The handling of the HTTP request that calls `report_data` should be (almost) stateless and generic.

As for the `generate_options` name: I am not going to merge this. As I said the name is **too generic in the context of the `ApplicationControler`**. What about `nested_list_options`? Does that make sense or is this method going to be relevant also for non-nested lists? I am sure you can figure out some better name.

Reply:
@martinpovolny is the concern here is purely about the name?